docs/source/java/vector.rst

```diff
@@ -268,6 +268,82 @@ For example, the code below shows how to build a :class:`ListVector` of int's us
+    VarCharVector decoded = (VarCharVector) encoder.decode(encoded)
+
+One thing we haven't discussed is how to create the dictionary vector from the original un-encoded values. That is left to the library user since a custom method will likely be more efficient than a general utility.
+
+Finally, you can package a number of dictionaries together, which is useful if you're working with a :class:`VectorSchemaRoot` with several dictionary-encoded vectors. This is done using an object called a :class:`DictionaryProvider`. as shown in the example below. Note that we don't put the dictionary vectors in the same :class:`VectorSchemaRoot` as the data vectors, as they will generally have fewer values.
+
+
+.. code-block:: Java
+
+    DictionaryProvider.MapDictionaryProvider provider =
+        new DictionaryProvider.MapDictionaryProvider();
+
+    provider.put(dictionary)))
+
+The :class:`DictionaryProvider` is simply a map of identifiers to :class:`Dictionary` objects, where each identifier is a long value. In the above code you will see it as the first argument to the :class:`DictionaryEncoding` constructor.
+
+This is where the :class:`DictionaryEncoding`'s 'id' attribute comes in. This value is used to connect dictionaries to instances of :class:`VectorSchemaRoot`, using a :class:`DictionaryProvider`.  Here's how that works:
+
+* The :class:`VectorSchemaRoot` has a :class:`Schema` object containing a list of :class:`Field` objects.
+* The field has an attribute called 'dictionary', but it holds a :class:`DictionaryEncoding` rather than a :class:`Dictionary`
+* As mentioned, the :class:`DictionaryProvider` holds dictionaries indexed by a long value. This value is the id from your :class:`DictionaryEncoding`.
+* To retrieve the dictionary for a vector in a :class:`VectorSchemaRoot`, you get the field associated with the vector, get its dictionary attribute, and use that object's id to look up the correct dictionary in the provider.
+
+.. code-block:: Java
+
+    // create the encoded vector, the Dictionary and DictionaryProvider as discussed above
+
+    // Create a VectorSchemaRoot with one encoded vector
+    VectorSchemaRoot vsr = new VectorSchemaRoot(List.of(encoded));
+
+    // now we want to decode our vector, so we retrieve its dictionary from the provider
+    Field f = vsr.getField(encoded.getName());
+    DictionaryEncoding encoding = f.getDictionary();
+    Dictionary dictionary = provider.get(encoding.getId());
+
+As you can see, a :class:`DictionaryProvider` is handy for managing the dictionaries associated with a :class:`VectorSchemaRoot`. More importantly, it helps package the dictionaries for a :class:`VectorSchemaRoot` when it's written. The classes :class:`ArrowFileWriter` and :class:`ArrowStreamWriter` both accept an optional :class:`DictionaryProvider` argument for that purpose. You can find example code for writing dictionaries in the documentation for (:ref:`ipc`).
```

Comment:
Also, the reader classes implement DictionaryProvider themselves

Reply:
Worth linking to the cookbook here? (Should the cookbooks be added via intersphinx for easier linking?)

Reply:
I don't think there's any information in the cookbook about dictionary encoding, although there is a GitHub issue on the subject

Reply:
I'm not that familiar with the readers but I can add a statement to that effect (although it seems odd that a reader would implement provider. I thought they were related one-to-one with a vector)

Reply:
No, dictionaries are disconnected from vectors, and DictionaryProvider is just a lookup table from dictionary ID to dictionary. Dictionaries are stored separately from batches in the IPC format. (C Data Interface muddles this: dictionaries are attached to vectors there.)

Reply:
Sorry, I wasn't suggesting they were linked directly to vector (and I was wrong about 1-1), but I believe a vector can be encoded by at most one dictionary. So my question is, if a Reader reads one vector, why does it need a relationship to a Map of dictionaries, rather than a single dictionary.

Reply:
Readers/writers read/write RecordBatches, not Vectors.

Reply:
I took "readers" to mean FieldReaders, but I realized later you meant ArrowReaders.